### ISSUE TITLE
Introducing Kotlin and Coroutines with MVVM

### ISSUE THREAD

**Issue by codedsun** [fossasia/phimpme-android#2903]
**Describe the feature you'd like**
The app currently needs to have following

- Kotlin 
- Async task and all DB related operations to Kotlin Coroutine
- Moving the architecture to MVVM

**Additional context**
This is a big task and will be taken by creating child issues and sending a PR. This will help to remove most of the bugs and clean the code with the operations of DB, Async task more smooth with coroutines. 

**Would you like to work on the issue?**
- [ ] Yes
- [ ] No
- Other: Open to All

**Comment by codedsun** [comment]
@yashk2000 @iamareebjamal  We are moving in this direction. Before moving to new  `opencamera` module version. I would like to do this prior so that all the app bugs are removed and app is more fast.

**Comment by iamareebjamal** [comment]
Any refactoring will inadvertently introduce more bugs, and not remove them. There are more important things to be done

**Comment by codedsun** [comment]
@iamareebjamal - Like? Except Fdroid issue

**Comment by codedsun** [comment]
@iamareebjamal There are bugs due to the code being written in not proper architecture. Before porting to new version of open-camera. I wanted this to be done.

**Comment by codedsun** [comment]
@iamareebjamal If you want the other things to be done, please list it out

**Comment by codedsun** [comment]
@iamareebjamal If you want the other things to be done, please list it out.
Currently the issue tracker has less than 25 issues, and more of them are features instead of bugs.

Also I would like to move to a good UI of the app since the project has same UI since the last 2 yrs. I have these goals in mind for the app.

- Kotlin+ MVVM
- UI improvement
- Open camera module to new version
- Phimpme remove accent color, theme activity
- Phimpme align to google photos
- Coroutines
- Fdroid

This is my task todo for this app. Please suggest what should be done instead

**Comment by iamareebjamal** [comment]
Fdroid + OpenCamera

Mark my words that any refactoring done by even the greatest software engineer leads to bugs. Refactoring is done to improve maintainability and design of code, bugs solved due to it are side effects. And as side effects, more bugs are introduced as well.

I cannot merge anything related to refactoring until a new release is made, Doing so will mean releasing a more buggy app than there is.

And I'm sorry, `moving to coroutines will speed the app speed with a x times of what it's currently.`, this claim is baseless. I am all in favour of coroutines, and they make stuff easy to write and maintain, but they don't make your app magically faster, they are helpful in structured concurrency. Anything synchronous will remain synchronous and anything asynchronous (AsyncTask) will remain async.

**Comment by codedsun** [comment]
@iamareebjamal MVVM? shall it be done or not

1. For Fdroid related task, I used mac system and as suggested I dont have a linux base. Nd my mackbook is 128GB so making a VM and installing the studio,etc will take space which is not available. I am trying to find the alternative.

2. Till the meantime, shall I go with
> Coroutines + MVVM 
or 
> Just take the new task of opencamera module bump

As the opencamera requires a lot of rewrite and refactor, I wanted to have this architecture and then do the bump of opencamera. 
But now, please guide me what shall I do and take it up for the codeheat is till February and I want to work on this.

Also for the UI and themed activity, do give the input.

Thanks for the detailed input for the above comments

**Comment by codedsun** [comment]
@iamareebjamal inputs please

**Comment by iamareebjamal** [comment]
Done after release on F-Droid. And integration of OpenCamera will not be easier or more difficult after you move to refactor, so do it first. If you think it'll be easier, please point to how. I think the things required to change will be same before and after the refactor

**Comment by codedsun** [comment]
@iamareebjamal Okay I am taking the opencamera bump issue, also for the Fdroid I am having storage issues in my mac, you suggested me that fdroidserver on mac has issues, so  I am finding an alternative for that.

After i bump the opencamera, I will be going with these issues of MVVM, Coroutines etc. Thanks a lot

**Comment by farhazulMullick** [comment]
@codedsun  I am a beginner in contributing to open source, how can i contribute in this project
